tests/gtmain.cpp

```diff
@@ -0,0 +1,131 @@
+/*++
+
+  INTEL CONFIDENTIAL
+  Copyright 2016 - 2017 Intel Corporation
+
+  The source code contained or described  herein and all documents related to
+  the  source  code  ("Material")  are  owned  by  Intel  Corporation  or its
+  suppliers  or  licensors.  Title   to  the  Material   remains  with  Intel
+  Corporation or  its suppliers  and licensors.  The Material  contains trade
+  secrets  and  proprietary  and  confidential  information  of Intel  or its
+  suppliers and licensors.  The Material is protected  by worldwide copyright
+  and trade secret laws and treaty provisions. No part of the Material may be
+  used,   copied,   reproduced,   modified,   published,   uploaded,  posted,
+  transmitted,  distributed, or  disclosed in  any way  without Intel's prior
+  express written permission.
+
+  No license under any patent, copyright,  trade secret or other intellectual
+  property  right  is  granted to  or conferred  upon  you by  disclosure  or
+  delivery of the  Materials, either  expressly, by  implication, inducement,
+  estoppel or otherwise. Any license  under such intellectual property rights
+  must be express and approved by Intel in writing.
+
+  --*/
+
+#include <unistd.h>
+#include <iostream>
+#include <getopt.h>
+#include <json-c/json.h>
+#include <common_test.h>
+#include <types_int.h>
+#include "gtest/gtest.h"
+
+using ::testing::TestCase;
+using ::testing::TestInfo;
+using ::testing::TestPartResult;
+using ::testing::UnitTest;
+
+using namespace std;
+using namespace common_test;
+
+void clean() {
+	for (auto& item : config_map) {
+		free(item.second);
+	}
+}
+
+struct option longopts[] = {{"help", no_argument, NULL, 'h'},
+                            {"bus", required_argument, NULL, 'b'},
+                            {"virt", no_argument, NULL, 'v'},
+                            {"json", required_argument, NULL, 'j'},
+                            {"plaform", required_argument, NULL, 'p'},
+                            {0, 0, 0, 0}};
+
+void show_help() {
+	cout << "Usage: gtapi" << endl
+	     << endl
+	     << "\t-b,--bus <bus> number." << endl
+	     << "\t-v,--virt <VM> disable auto-reconfiguration of NLB0." << endl
+	     << "\t-j,--json <path> provide path to bitstream json file." << endl
+	     << "\t-p,--platform <ase/fpga> select which platform to run test."
+	     << endl
+	     << endl;
+}
+
+int main(int argc, char** argv) {
+	int getopt_ret;
+	int option_index;
+
+	const char* path;
+	path = "../tests/configuration.json";
```

Comment:
I'd recommend making this a command line argument defaulted to "configuration.json" so that it will look for the config file in $PWD/configuration.json? We just need to:
* Copy/configure file to `${CMAKE_BINARY_DIR}`
* Run it from `${CMAKE_BINARY_DIR}`

We can also embed the default configuration in a JSON string in the binary.